### PR TITLE
feat(content): Auxiliary Mark II

### PR DESCRIPTION
**Pull request by LorenzoBolla**
## Summary
The Auxiliary currently does not have a Mark II variant. At first glace, that's reasonable - a logistical ship does not need top of the line and expensive engines and weapons. It also does not have the space to easily upgrade them, anyways. However, the Auxiliary uses a Ramscoop, and the Catalytic Ramscoop is equivalent to 7 normal Ramscoops, for just 6 tons more space. Since the Auxiliary has 57 tons of outfit space leftover, there's little reason not to upgrade, it simply makes the Auxiliary a lot better at long range logistical work for very little cost.

This PR adds a Mark II variant to all 3 Auxiliaries and adds them to fleets when the Catalytic Ramscoop becomes available commercially for civilians. It also changes the Auxiliary that can appear in the "Human Wormhole Guards" fleet during Hai Reveal to Mark II, and changes the Auxiliary you can attack during the Wolf Pack missions to a Mark II variant as well. Plundering and scanning auxiliaries after that event should now lead to a cool little "Oh! Yeah, of course they use those now." moment for the player, at least if they're like me and like to see different parts of the game interacting with one another. 

## Testing

Worked through the FW campaign until i found the Catalytic Ramscoop event trigger and have tested it appropriately. As of 2023-02-18, it works perfectly and is ready to be merged.

Save file, if anybody wishes to test it themselves; just scan an auxiliary:
[Rebbia_Rampart3025-01-31_Officially_Signing_Up_with_FW~ONE DAY TO CAT SCOOP RELEASE!.txt](https://github.com/endless-sky/endless-sky/files/10854069/Rebbia_Rampart3025-01-31_Officially_Signing_Up_with_FW.ONE.DAY.TO.CAT.SCOOP.RELEASE.txt)

## To-do

- [x] Test it
- [x] ~~Make it actually work. It seems not to, in my testing. I don't know what's wrong with the code, so any help is appreciated.~~ Turns out i was testing it wrong! Should work, but:
- [x] Test it the right way (save file provided by Zitchas, from start of FW, will update when i've gotten to the correct point in the story to trigger the event)

### CODE REVIEW

File: data/human/events.txt [hunk 1]
```diff
@@ -1167,12 +1167,95 @@ event "navy occupying the south"
 event "catalytic ramscoop available"
 	outfitter "Deep Sky Basics"
 		"Catalytic Ramscoop"
-
-
-
-event "fw gets catalytic ramscoop"
```
Comment (Anarchist2) [inline data/human/events.txt:1173]: Why is this deleted?

File: data/human/events.txt [hunk 2]
```diff
@@ -1167,12 +1167,95 @@ event "navy occupying the south"
...
+			"Lance" 6
+			"Cruiser" 2
+			"Combat Drone" 8
+		add variant 1
+			"Auxiliary (Transport Mark II)" 3
+			"Dropship" 9
+			"Lance" 3
+			"Cruiser" 2
+			"Combat Drone" 8
+		remove variant
```
Comment (Anarchist2) [inline data/human/events.txt:1215]: The other event that updates Navy fleets doesn't remove variants.

File: data/human/events.txt [hunk 3]
```diff
@@ -1167,12 +1167,95 @@ event "navy occupying the south"
 event "catalytic ramscoop available"
 	outfitter "Deep Sky Basics"
 		"Catalytic Ramscoop"
-
-
-
-event "fw gets catalytic ramscoop"
-	outfitter "Delta V Advanced"
-		"Catalytic Ramscoop"
+	fleet "Republic Logistics"
```
Comment (Anarchist2) [inline data/human/events.txt:1170]: Why are the logistics fleets changed here instead of in `event "navy using mark ii ships"`?

File: data/human/events.txt (hunk 1 above)
Comment (LorenzoBolla) [reply data/human/events.txt:1173]: Dangit, i accidentally deleted it when i pasted over the changes. Thanks for pointing it out, let me add it back in.

File: data/human/events.txt (hunk 2 above)
Comment (LorenzoBolla) [reply data/human/events.txt:1215]: Due to economies of scale and standardisation, having these fully replace the old variants would reduce maintenance costs enough while adding such a large direct benefit to long range logistics, it would be justified to fully replace them instead of simply supplementing them, in my eyes. It is also a very small change as opposed to a nearly complete one, with the other Mark IIs.

I can change it so that it still only makes them as common as the old variants as opposed to replacing them, but, again, it's such a small and direct upgrade i don't see why to do that. This is my thought process behind this being different.

File: data/human/events.txt (hunk 3 above)
Comment (LorenzoBolla) [reply data/human/events.txt:1170]: No other Mark II ship uses Catalytic Ramscoops and the Catalytic Ramscoop becomes publicly available separately from the Mark II variant's introduction. If i'm not mistaken, only afterwards is the Catalytic Ramscoop introduced. To avoid the player finding the Catalytic Ramscoop too long before they receive the mission in which you steal the blueprints for the FW through a sympathiser, i choose to take this route. If it isn't actually conflicting in a significant way, i can simply swap them so they are further down, though. Or i could swap it further down anyways, as the navy may have gotten early access to the tech. 

What do you think?

Comment (Anarchist2) [reply data/human/events.txt:1170]: The Catalytic Ramscoop is made available before Mark II ships are introduced.

File: data/human/events.txt (hunk 2 above)
Comment (Anarchist2) [reply data/human/events.txt:1215]: I'd rather avoid overcomplicating things, but if you think that the extra overhead is worth it, go ahead.

File: data/human/variants.txt [hunk 4]
```diff
@@ -97,6 +97,69 @@ ship "Arrow" "Arrow (Hai)"
...
+		"Large Radar Jammer"
+		"Catalytic Ramscoop"
+		
+		"X4700 Ion Thruster"
+		"X1700 Ion Thruster"
+		"X5200 Ion Steering"
+		"Scram Drive"
+
+
+ship "Auxiliary (Cargo)" "Auxiliary (Cargo Mark II)"
```
Comment (Anarchist2) [inline data/human/variants.txt:121]: The game doesn't like variants of variants. You'll have to partially repeat the Auxiliary (Cargo) variant definition.

File: data/human/events.txt (hunk 3 above)
Comment (LorenzoBolla) [reply data/human/events.txt:1170]: Then it is best to keep these separate, so they show up sooner, as that's when they are made available. Thank you for the help!

File: data/human/variants.txt (hunk 4 above)
Comment (LorenzoBolla) [reply data/human/variants.txt:121]: Thank you, i was trying to figure out why the check failed! Would it be best to separate it like this: 
```suggestion
ship "Auxiliary (Cargo)" "Auxiliary (Cargo) (Mark II)"
```
Or some other way? Or do you mean i'd have to repeat something else?

Comment (LorenzoBolla) [reply data/human/variants.txt]: Tried it and it seems happy... alright.

Comment (Anarchist2) [reply data/human/variants.txt]: Doing that will cause the model name of the Auxiliary to show up as "Auxiliary (Cargo)". You have to do something like this instead:

https://github.com/endless-sky/endless-sky/blob/a0fa9c54d17c7a868102204e512936478ba87039/data/human/variants.txt#L100-L135

Comment (LorenzoBolla) [reply data/human/variants.txt]: Ah, i see, i need to add the atributes like the variants do so on ships.txt on variants.txt as well. Thank you so much for the help!

**Comment by LorenzoBolla** [comment]
Alright, changing which event triggers it doesn't do it, so it isn't that it's triggered by an event that also changes outfitters. I can't really think of much else - if it was the variants-of-variants issue, the normal Auxiliary would have been changed at least, so it isn't that either. I double checked and i wrote the fleet name correctly, so that isn't the issue, either. 

I'm at a loss as to what is the issue with these variants not triggering. I'm testing them on a save with the event set as triggered, so Auxiliaries should show up with Catalytic Ramscoop in place of normal ones, but they do not. Does anybody have any ideas?

**Comment by LorenzoBolla** [comment]
Worked through the FW campaign until i found the Catalytic Ramscoop event trigger and have tested it appropriately. As of 2023-02-18, it works perfectly and is ready to be merged.

Save file, if anybody wishes to test it themselves:
[Rebbia_Rampart3025-01-31_Officially_Signing_Up_with_FW~ONE DAY TO CAT SCOOP RELEASE!.txt](https://github.com/endless-sky/endless-sky/files/10854063/Rebbia_Rampart3025-01-31_Officially_Signing_Up_with_FW.ONE.DAY.TO.CAT.SCOOP.RELEASE.txt)

**Comment by LorenzoBolla** [comment]
Re-requested your review as it now was tested and works appropriately. Again, thanks for helping me with the variants-of-variants thing!
